### PR TITLE
Add bad file extension error message

### PR DESCRIPTION
**Pull request by ghost**
ntdkhiem and my attempt at blocking the user from opening forbidden file extensions. Currently sends an error message popup that sends user back to the text editing area; would be better to remain in scroll menu.

### CODE REVIEW

**Comment by ntdkhiem** [comment]
We should finish up the remaining tasks before actually merge this.

File: src/text/text_editor.py
Comment (BoraxTheClean) [inline src/text/text_editor.py:359]: Hell yes, nice comments!

File: tests/basic_tests.py
Comment (BoraxTheClean) [inline tests/basic_tests.py]: What does this do? Does it just facilitate inputing text with other tests? Maybe add some comments.

File: src/scroll/scroll_menu.py
Comment (BoraxTheClean) [inline src/scroll/scroll_menu.py]: Commenting this method and improving the docstring would be pretty dece.

**Comment by BoraxTheClean** [comment]
> We should finish up the remaining tasks before actually merge this.

Which Issues is this blocked by?

Long lived feature branches are kind of an anti-pattern. So we should be prioritizing merging over waiting whenever possible.

**Comment by ntdkhiem** [comment]
@BoraxTheClean , we're done. Could you take a look?

File: .gitignore
Comment (BoraxTheClean) [inline .gitignore]:
```suggestion
# ThoughtBox's Internal Application File System
```

File: src/scroll/scroll_menu.py
Comment (BoraxTheClean) [inline src/scroll/scroll_menu.py]:
```suggestion
            """Handles actions related to loading previews of files into the file explorer."""
```
Is this an accurate docstring for this function?

**Review by BoraxTheClean — APPROVED** [review]
This looks good. I think we should make a ticket for the backlog to clean up the code in:

`def set_done() -> None:` and similar functions. They seem to work nicely, but they are hacky and complex to reason about.

File: src/scroll/scroll_menu.py
Comment (ntdkhiem) [reply src/scroll/scroll_menu.py]: I think it should be `handles actions related to adding file's contents to text editor`

Comment (BoraxTheClean) [reply src/scroll/scroll_menu.py]: Oh, the code in `scroll_menu.py` is loading files into the text editor? That seems kind of backwards. Maybe add an issue to refactor? Definitely add that docstring too.

**Comment by ghost** [comment]
We completed the three features from the original issue:

1. Open the app in a directory called .`thought_box` if it exists, creating it if not.
2. Disable the .. button in the scrollbar menu in `.thought_box` itself to restrict the app to that directory and its children.
3. Prevent opening files with extensions other than `.txt` or `.md` and generate an error message if the user attempts it.

I noted two issues to fix:
1. The error message for loading a disallowed extension moves focus to the editing area from the scroll menu, instead of staying there.
2. The .. button doesn't appear for empty directories.

**Pull request merged**
